open-api/rest-catalog-open-api.yaml

```diff
@@ -3265,6 +3265,133 @@ components:
+              - If a listed column has a transform, the reader MUST apply it and replace
+                all references to the underlying column with the transformed value
+                (for example, truncate[4](cc) MUST be projected as truncate[4](cc) AS cc,
+                and all references to cc during query evaluation post applying required-row-filter MUST resolve to this alias).
+              - Columns not listed in the required-colum-projections MUST NOT be read.
+
+            2. A column MUST appear at most once in the required-colum-projections.
+      
+            3. If a projection entry includes an action that the reader cannot evaluate,
+              the reader MUST fail rather than ignore the transform.
+      
+            5. An identity transform is equivalent to projecting the column directly.
+            
+            8. The data type of the projected column MUST match the data type defined for the transform result.
+
+          type: array
+          items:
+            $ref: '#/components/schemas/Projection'
+        required-row-filter:
+          description: >
+            An expression that filters rows in the table.
+      
+            1. A reader MUST discard any row for which the filter evaluates to false or null, and
+              no information derived from discarded rows MAY be included in the query result.
+            
+            2. Row filters MUST be evaluated against the original, untransformed column values.
+              Required projections MUST be applied only after row filters are applied.
+      
+            3. If the catalog supports multiple row access filters for the table, it is
+              the catalog's responsibility to combine them using the appropriate logic (e.g., AND, OR).
+      
+            4. If a client cannot interpret or evaluate a provided filter expression, it MUST fail.
+      
+            5. If the required-row-filter property is absent or empty, no mandatory filtering is imposed.
+          $ref: '#/components/schemas/Expression'
+
+    Projection:
+      type: object
+      description: Defines a projection for a column.
+      properties:
+        field-id:
+          type: integer
+          description: field id of the column being projected.
+        action:
+          $ref: '#/components/schemas/Action'
+      required:
+        - field-id
+        - action
+
+    Action:
+      description: Defines the specific action to be executed for computing the projection.
+      oneOf:
+        - $ref: '#/components/schemas/MaskHashSha256'
+        - $ref: '#/components/schemas/MaskReplaceWithNull'
+        - $ref: '#/components/schemas/MaskAlphanumeric'
+        - $ref: '#/components/schemas/ApplyTransform'
+
+    MaskHashSha256:
+      type: object
+      description: Mask the data of the column by apply SHA256 hash algorithm. Engines are free to use their own implementation of SHA256.
```

Comment:
Either this should be any cryptographic hash algorithm, or this should specify how to calculate values so they are consistent across engines. I don't really care which one, but if we are specifying details then we should specify all of them.

Reply:
rephrase it with the spec definition of sha256 please let me know if this reads better ?